controller/src/main/java/org/jboss/as/controller/CapabilityRegistry.java

```diff
@@ -385,11 +385,12 @@ private RuntimeStatus getCapabilityStatus(CapabilityId id, Set<CapabilityId> exa
             if (reloadCapabilities.contains(id)) {
                 return RuntimeStatus.RELOAD_REQUIRED;
             }
-            examined.add(id);
         } // else defer reload-required check until after we search requirements for restart-required
 
+        visited.add(id);
```

Comment:
This is the only relevant change, move it outside of the previous if to make it count when we have capabilities in the `restartCapabilities`. I'm not sure what was the original intention, looks like it was expected to have it in the `restartCapabilities` array, but that's not always true. I renamed it to `visited` instead of `examined,` since examined could mean already checked its status, which is not correct at this point. At the end of this method, the capability is checked (examined) to see if it is in the `reloadCapabilities.` I found visited more adequate.